### DISCUSSION TITLE
Simulation prior to physical install. Possible?

### DISCUSSION THREAD

**Discussion by twack** [aqualinkd/AqualinkD#273] [Q&A] [answered]
I'd like to configure and test as much as possible prior to recieving the 485 and physically hooking it up. Is it possible? 

Thanks in advance, for any responses.

**Comment by sfeakes** [comment, marked as answer]
Not really.  It’s dependent on a tty serial port being available. You can plug any old USB adapter with serial chip in the raspberry that makes the port available. (Ie any hardware to get /dev/ttyUSB0) Then AqualinkD will start and you could do a few things, but it will quickly fail if it doesn’t read anything on the serial port after 30 seconds or so.

**Reply by twack** [reply]
I'm going to use the waveshare USB/ETH hat on the pi zero 2w. Is that or something simular what ppl are using? 

This one: https://www.amazon.com/USB-HUB-HAT-Board-Compatible/dp/B07X1BH5FN

**Reply by chrisdaun** [reply]
@twack that waveshare hat won't give you tty-to-rs485.

I tried waveshare's rs485 can hat but it only interacts with the Aqualink bus at 3.3v. I needed 5v so this didn't end up working for me.
https://a.co/d/2uhzVop

I ended up using this and powering it via the 5v gpio pins on an orangepi
https://a.aliexpress.com/_mKhV6z4
